src/jit/hwintrinsicxarch.cpp

```diff
@@ -337,44 +467,87 @@ GenTree* Compiler::impX86HWIntrinsic(NamedIntrinsic        intrinsic,
     {
         return gtNewIconNode(issupported);
     }
-    else if (!issupported)
+    // - calling to unsupported intrinsics must throw PlatforNotSupportedException
+    // - 64-bit (r64) intrinsics must throw PlatforNotSupportedException on 32-bit platform
+    else if (!issupported || is64bitOnlyIntrinsicOn32bit(category, callType))
     {
         return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
     }
 
+    // table-driven importer of simple intrinsics
+    if (impTableDrivableHWIntrinsic(category))
+    {
+        unsigned int sizeBytes;
+        var_types    baseType = getBaseTypeAndSizeOfSIMDType(sig->retTypeSigClass, &sizeBytes);
+        assert(baseType != TYP_UNKNOWN && sizeBytes != 0);
+        var_types retType  = getSIMDTypeForSize(sizeBytes);
+        unsigned  simdSize = simdSizeOfHWIntrinsic(intrinsic, sig);
+
+        assert(numArgs > 0);
+        assert(retType != TYP_UNDEF);
+        assert(retType == TYP_SIMD16 || retType == TYP_SIMD32);
+        assert(instructionOfHWIntrinsicOnType(intrinsic, baseType) != INS_invalid);
+        assert(simdSize == 32 || simdSize == 16);
+
+        GenTree* retNode = nullptr;
+        GenTree* op1     = nullptr;
+        GenTree* op2     = nullptr;
+
+        switch (numArgs)
+        {
+            case 1:
+                op1     = impPopStack().val;
+                retNode = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, baseType, simdSize);
```

Comment:
There needs to be some validation of the operand types that are popped off the stack. Is that being done elsewhere?

Reply:
I will extend this part (1-arg table-driven intrinsics) with `Load*`, so I will add validation in that PR.

Reply:
The old code had some validation with `impSIMDPopStack`, it looks like that was lost in the new code so that all intrinsics could share the same code path.

I think something similar to the old code might be more desirable, as it can continue having validation logic on the parameters and signature (but still lets us join a large number of code paths).

It might be worth having the table carry additional information in Checked/Debug builds (signature information) and to have a `ValidateSignature` helper function which asserts the signature we have for the intrinsic matches what we expect (for all args and the return value).